src/request_ext.rs

```diff
@@ -0,0 +1,136 @@
+use crate::config::ConfigBuilder;
+use crate::typestate::HttpCrateScope;
+use crate::{http, Agent, AsSendBody};
+
+/// Extension trait for [`http::Request<impl AsSendBody>`].
```

Comment:
I've tried to match the style of documentation and examples to the existing styles, but lmk if you want any changes made.